lib/releases.js

```diff
@@ -29,14 +32,16 @@ class Release {
  */
 async function fetchReleases () {
   const url = 'https://api.github.com/repos/opentofu/opentofu/releases';
+  const githubToken = core.getInput('github_token');
+  const is3rdPartyRunner = process.env.FORGEJO_ACTIONS || process.env.GITEA_ACTIONS;
 
   const headers = {
     Accept: 'application/vnd.github+json',
     'X-GitHub-Api-Version': '2022-11-28'
   };
 
-  if (process.env.GITHUB_TOKEN) {
-    headers.Authorization = `Bearer ${process.env.GITHUB_TOKEN}`;
+  if (githubToken === '' && !is3rdPartyRunner) {
+    headers.Authorization = `Bearer ${githubToken}`;
```

Comment:
This won't work on GitHub since the `githubToken` will only be used if it is empty and the environment variable is not read at all.

Reply:
oh yes, indeed